src/lib/edit_script.rs

```diff
@@ -134,43 +154,57 @@ impl Chawathe96Config {
                                                  store: &MappingStore<T>,
                                                  x: NodeId,
                                                  new_mappings: &TemporaryMappingStore,
+                                                 from_in_order: &HashSet<NodeId>,
                                                  to_in_order: &HashSet<NodeId>)
                                                  -> u16 {
-        debug!("find_pos({})", x);
-        if x.is_root(&store.to_arena) {
-            return 0;
-        }
+        // 1. Let y=p(x) in T_2 and let w be the partner of x in T_1.
+        // N.B. w seems to be unused in the algorithm.
         let y = store.to_arena[x].parent().unwrap();
+        // 2. If x is the leftmost child of y that is marked "in order" return
+        // 0 (the paper says 1 but we count child nodes from 0).
         let siblings = y.children(&store.to_arena).collect::<Vec<NodeId>>();
         for child in &siblings {
             if to_in_order.contains(child) {
                 if x == *child {
+                    debug!("find_pos({}) <- 0. Node in order.", x);
                     return 0;
                 } else {
                     break;
                 }
             }
         }
         let x_pos = x.get_child_position(&store.to_arena).unwrap();
-        // Find v in T2, where v is the rightmost sibling of x that is to the
-        // left of x and is marked "in order".
+        // 3. Find v in T_2, where v is the rightmost sibling of x that is to
+        // the left of x and is marked "in order".
         let v = x.children(&store.to_arena)
             .take(x_pos)
             .filter(|c| to_in_order.contains(&c))
             .last();
         if v.is_none() {
-            return 0; // No right-most sibling in order.
+            debug!("find_pos({}) <- 0. No right-most sibling in order.", x);
+            return 0;
         }
-        let u = new_mappings.get_from(&v.unwrap()).unwrap();
-        let u_pos = u.get_child_position(&store.to_arena).unwrap();
-        u_pos as u16 + 1
+        // 4. Let u be the partner of v in T_1.
+        let u_opt = new_mappings.get_from(&v.unwrap());
+        assert!(u_opt.is_some(), "find_pos() could not find a partner for v");
+        let u = u_opt.unwrap();
+        // 5. Suppose u is the i'th child of its parent that is marked
+        // "in order". Return i + 1.
+        let u_p = store.from_arena[u].parent().unwrap();
+        let u_pos = u.get_child_position(&store.from_arena).unwrap();
+        let ret = u_p.children(&store.from_arena)
+            .take(u_pos)
```

Comment:
Indentation doesn't line up.

Reply:
`rustfmt` doesn't change this file, and that `.take` is 4 spaces to the right of the `l` in the `let`... ?